### ISSUE TITLE
[GraphTrainer] Loss region FX nodes missing stacktrace and module_fqn on some backward nodes

### ISSUE THREAD

**Issue by SherlockNoMad** [pytorch/torchtitan#3208]
## Summary

After annotating the loss function with `module_fqn: 'loss'` via `@annotate_fn` (#3207), most loss-region nodes now carry the annotation correctly. However, two upstream issues remain:

### 1. Missing stacktrace on loss-region nodes

Nearly all forward and backward nodes in the loss region show `No stacktrace found for following nodes`. The loss function (`cross_entropy_loss`) is a plain function (not an `nn.Module`), so `record_stack_traces=True` in `make_fx` doesn't capture source locations for its decomposed ops. Only DTensor wrapper ops (e.g. `view_as` from `torch/distributed/tensor/_api.py`) have stacktraces.

### 2. Backward nodes at `seq_nr: 1174` missing `module_fqn` annotation

`_copy_fwd_metadata_to_bw_nodes` copies `module_fqn` from forward to backward nodes by matching `seq_nr`. However, there are no forward nodes with `seq_nr: 1174` — only backward nodes at that `seq_nr` exist (the cross-entropy backward through DTensor). Without a forward counterpart, the copy pass has nothing to propagate, so these backward nodes remain unannotated.

This appears to be a PyTorch autograd `seq_nr` assignment issue where the backward ops get a `seq_nr` that doesn't correspond to any forward node.

### Affected nodes

**Forward** (all annotated correctly):
- `seq_nr: 1171-1173` — `view`, `_to_copy`, cross-entropy decomposition (amax, sub, exp, sum, log, etc.)
- `seq_nr: 1175` — DTensor `view_as`, nll_loss decomposition (ne, gather, neg, etc.)
- `seq_nr: 1176` — `div` (loss / global_valid_tokens)

**Forward** (missing annotation):
- `seq_nr: 1176` — `ones_like_1` (autograd grad_output seed, generated outside annotated scope)

**Backward** (missing annotation due to no forward match):
- `seq_nr: 1174` — ~20 ops: DTensor view, unsqueeze, ne, where, zeros_like, index_put, exp, add, mul (cross-entropy backward)

### Both are upstream PyTorch issues
These cannot be fixed from the GraphTrainer annotation side. They would need changes in:
- PyTorch's `make_fx` stack trace recording for non-module callables
- PyTorch's autograd `seq_nr` assignment to ensure backward nodes have matching forward counterparts

**Comment by SherlockNoMad** [comment]
Sample Tlp: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpDKei0I/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000

**Comment by ydwu4** [comment]
**p2:**

Minimal repro: P2305025425

The cause: seq_nr=1174 is the forward-created autograd NllLossBackward0 node from DTensor loss_parallel cross entropy. DTensor intercepts/decomposes the visible [nll_loss_forward](https://github.com/pytorch/pytorch/blob/main/torch/distributed/tensor/parallel/loss.py#L501) , so the high-level autograd seq
has no matching forward FX node (because the seq_nr is occupied when dispatching autograd key of nll_loss_forward). The graph only consists of decomposed ops.:
```python
  forward executes cross_entropy on DTensor

  aten._log_softmax creates LogSoftmaxBackward0
    -> seq_nr 1173
    -> traced forward decomposition ops get seq_nr 1173

  aten.nll_loss_forward creates NllLossBackward0
    -> seq_nr 1174
    -> but DTensor custom handler decomposes the visible forward work
    -> handler-body FX ops show up later, e.g. seq_nr 1175

  final loss div creates DivBackward0
    -> seq_nr 1176

  Then backward runs from the actual autograd graph:

  DivBackward0 seq_nr 1176
    -> NllLossBackward0 seq_nr 1174
    -> LogSoftmaxBackward0 seq_nr 1173
```

The lesson is that when dtensor has customized decompositions, there will be dangling seq_nr in backward.

One way to address this is just map 1174 to nodes that are after 1173 and above 1176.
